src/Conversion/ONNXToTorch/NN/FlattenOp.cpp

```diff
@@ -129,35 +126,39 @@ class ONNXFlattenOpToTorchLowering : public ConversionPattern {
     // What we feel is that there will be two steps required -
     //
     //  1) flatten the region from 0 position to axis - 1.
-    //  2) flatten the region from axisValue to inputRank( last dimension).
+    //  2) Since all dimensions before `axis` have already been 
+    //     condensed into a single one (dim 0),
+    //     we set start=1. We use -1 as the end value, which tells torch
+    //     to go until the last dimension.
     /********************************************************************/
 
 
     if (axisValue > 1) {
-      // Flatten the region upto axis-1.
-
       // Build the intermediate result type.
-      // This is the same type as the input, with all dims before the axis value collapsed into one.
+      // This is the same type as the input, with all dims before the axis
+      // value collapsed into one.
       ArrayRef<int64_t> inputShape = inputTensorType.getShape();
       auto numDimsAfterAxis = inputShape.size() - axisValue;
       auto remainingDims = inputShape.take_back(numDimsAfterAxis);
       std::vector<int64_t> intermShape;
-      intermShape.push_back(resultTensorType.getShape()[0]); // this is the collapsed dimension
-      intermShape.insert(intermShape.end(), remainingDims.begin(), remainingDims.end());
+      // this is the collapsed dimension
+      intermShape.push_back(resultTensorType.getShape()[0]);
+      intermShape.insert(intermShape.end(), remainingDims.begin(), 
+		remainingDims.end());
       auto intermType = Torch::ValueTensorType::get(context,
-       llvm::makeArrayRef(intermShape), // todo is this a memory leak? where to allocate the vector?
-       inputTensorType.getElementType());
-
-      result = createAtenFlattenOp(
-          rewriter, loc, result, intermType, /*start=*/0, /*start=*/axisValue - 1, op1);
+       		llvm::makeArrayRef(intermShape), // todo is this a memory 
+		//leak? where to allocate the vector?
```

Comment:
Nit: Put this comment line at the end of the next code line.

`llvm::makeArrayRef(intermShape),      // todo is this a memory`
`  inputTensorType.getElementType());  // leak? where to allocate the vector?`

Reply:
Actually the comment can be removed. I looked into it and it's not a memory leak, the tensor type has its own internal storage and this is safe.

Reply:
Removed the comment.